ibc/modules/src/applications/transfer/acknowledgement.rs

```diff
@@ -13,66 +13,49 @@
 // limitations under the License.
 
 use super::error::Error;
-use crate::{core::ics26_routing::context::Acknowledgement as AckTrait, prelude::*};
+use crate::prelude::*;
 use core::fmt::{Display, Formatter};
 
-use serde::{Deserialize, Deserializer};
+use serde::{Deserialize, Serialize};
 
 /// A string constant included in error acknowledgements.
 /// NOTE: Changing this const is state machine breaking as acknowledgements are written into state
 pub const ACK_ERR_STR: &str = "error handling packet on destination chain: see events for details";
-pub const ACK_SUCCESS_B64: &[u8] = b"AQ==";
+pub const ACK_SUCCESS_B64: &str = "AQ==";
 
-#[derive(Clone, Debug)]
+#[derive(Serialize, Deserialize, Debug, Clone)]
```

Comment:
@vmarkushin Does this still serialize into a json object?

Reply:
Yes, I've checked

Reply:
The variant name is the key in JSON:
`Ack::Result` -> `{ "result": "..." }`
`Ack::Error` -> `{ "error": "..." }`

Reply:
Okay cool